scaffolding-chef/lib/scaffolding.sh

```diff
@@ -0,0 +1,129 @@
+#
+# A scaffolding for Chef Policyfile packages
+#
+
+scaffolding_load() {
+  pkg_deps=("adam/chef-dk" "core/cacerts" ${pkg_deps[@]})
+  pkg_build_deps=("adam/chef-dk" "core/git" ${pkg_build_deps[@]})
+  pkg_svc_user=("root")
+
+  if [ -z "${scaffold_policy_name+x}" ]; then
+    echo "You must set \$scaffold_policy_name to a valid policy name. For example:"
+    echo
+    echo "\$scaffold_policy_name=base"
+    echo
+    echo "Will build a base.rb policyfile"
+    exit 1
+  fi
+}
+
+do_default_download() {
+  return 0
+}
+
+do_default_verify() {
+  return 0
+}
+
+do_default_unpack() {
+  return 0
+}
+
+do_default_build() {
+  if [ -d "$PLAN_CONTEXT/../policfyfiles" ]; then
+    _policyfile_path="$PLAN_CONTEXT/../policyfiles"
+  else
+    if [ -d "$PLAN_CONTEXT/../../policyfiles" ]; then
+      _policyfile_path="$PLAN_CONTEXT/../../policyfiles"
+    else
+      if [ -d "$PLAN_CONTEXT/../../../policyfiles" ]; then
+        _policyfile_path="$PLAN_CONTEXT/../../../policyfiles"
+      else
+        echo "Cannot detect a policyfiles directory!"
+        exit 1
+      fi
+    fi
+  fi
+  rm -f $_policyfile_path/*.lock.json
+  policyfile=$_policyfile_path/$scaffold_policy_name.rb
+  for x in `cat $policyfile | grep include_policy | awk -F "," '{print $1}' | awk -F '"' '{print $2}' | tr -d " "`; do
```

Comment:
Don't the tools do this for you? or is this to ensure that we have the sub policy in the same dir, rather than remotely?

Reply:
We tested this, and we don't compile policies specified by include_policy before trying to include them - so this is attempting to make sure that (as long as they're specified in the right order) we also compile policies in order.